### PR TITLE
Add auth logic

### PR DESCRIPTION
**Pull request by ikoenigsknecht**

### CODE REVIEW

File: demos/quiet-sandbox/package.json [hunk 1]
```diff
@@ -16,23 +15,30 @@
...
   "author": "",
   "license": "ISC",
+  "type": "module",
   "dependencies": {
-    "@localfirst/auth": "workspace:*",
     "@commander-js/extra-typings": "^12.1.0",
     "@inquirer/core": "^9.0.2",
-    "@types/figlet": "^1.5.8",
+    "@inquirer/prompts": "^5.3.0",
+    "@inquirer/type": "^1.5.1",
```
Comment (ikoenigsknecht) [inline demos/quiet-sandbox/package.json:23]: I had to add this stuff to get the build to work, not sure why.

File: demos/quiet-sandbox/scripts/test_auth.ts [hunk 2]
```diff
@@ -0,0 +1,91 @@
+#! /usr/bin/env ts-node
```
Comment (ikoenigsknecht) [inline demos/quiet-sandbox/scripts/test_auth.ts:1]: This is the code that I was using in my `index.ts` to test the logic out.  This should give an idea of how my code is used.

**Review by adrastaea — CHANGES_REQUESTED** [review]

File: demos/quiet-sandbox/src/auth/chain.ts [hunk 3]
```diff
@@ -0,0 +1,107 @@
+/**
+ * Handles generating the chain and aggregating all chain operations
+ */
+
+import * as auth from '@localfirst/auth'
+import { LoadedSigChain } from './types.js'
+import { UserService } from './services/members/user_service.js'
```
[the remark this answers is not in the document]
Comment (ikoenigsknecht) [reply demos/quiet-sandbox/src/auth/chain.ts]: Done.

File: demos/quiet-sandbox/src/auth/services/invites/invite_service.ts [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+      expiration,
+      maxUses
+    })
+    this.activeSigChain.persist()
+    return invitation
+  }
+
+  public revoke(id: string) {
+    this.activeSigChain.team.revokeInvitation(id)
+    this.activeSigChain.persist()
```
[the remark this answers is not in the document]
Comment (ikoenigsknecht) [reply demos/quiet-sandbox/src/auth/services/invites/invite_service.ts]: Done.

**Review by adrastaea — APPROVED** [review]

**Pull request merged**